pom.xml

```diff
@@ -762,5 +780,41 @@
                 </pluginManagement>
             </build>
         </profile>
+
+        <profile>
+            <id>Java 16+</id>
+            <activation>
+                <jdk>[16,)</jdk>
+            </activation>
+            <build>
+                <pluginManagement>
+                    <plugins>
+                        <plugin>
+                            <groupId>org.apache.maven.plugins</groupId>
+                            <artifactId>maven-compiler-plugin</artifactId>
+                            <configuration>
+                                <!-- Flags needed for errorprone -->
+                                <!-- On JDK 16 and newer, additional flags are
+                                required due to JEP 396: Strongly Encapsulate
+                                JDK Internals by Default.
+                                See https://errorprone.info/docs/installation -->
+                                <compilerArgs combine.children="append">
```

Comment:
I'm OK with this, as long as we don't need the internals for our code.

Reply:
It is strictly for error prone to work, which is only used at compile time for developers to ensure best practices are being followed.